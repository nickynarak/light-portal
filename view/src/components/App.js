import React from 'react';
import PropTypes from 'prop-types';

const ContextType = {
  // Redux store
  store: PropTypes.object.isRequired,
  //routes: PropTypes.object.isRequired,
};

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context;
  }

  render() {
    // NOTE: If you need to add or modify header, footer etc. of the app,
    // please do that inside the Layout component.
    return React.Children.only(this.props.children);
  }
}

export default App;
