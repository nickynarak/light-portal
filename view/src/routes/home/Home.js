import React, { Component } from 'react';
import Layout from '../../components/Layout/Layout';

class Home extends Component {
  render() {
    return (
      <Layout>
        <p>Welcome to light portal</p>
      </Layout>
    );
  }
}

export default Home;
