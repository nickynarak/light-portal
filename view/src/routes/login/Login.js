import React, { Component } from 'react';
import Layout from '../../components/Layout/Layout';

class Login extends Component {
  render() {
    return (
      <Layout>
        <p>Login Page Here</p>
      </Layout>
    );
  }
}

export default Login;
