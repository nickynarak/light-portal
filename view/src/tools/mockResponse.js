/**
 * Returns a mock response from server
 * @param status {integer} - Status server code
 * @param statusText {string} - Server text
 * @param response {string} - Server return in strings
 * @returns Response object
 */
export default (status, statusText, response) =>
  new window.Response(response, {
    status,
    statusText,
    headers: {
      'Content-type': 'application/json',
    },
  });
