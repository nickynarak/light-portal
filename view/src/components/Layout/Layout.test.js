/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import mockResponse from '../../tools/mockResponse';
import App from '../App';
import Home from '../../routes/home/Home';
import Login from '../../routes/login/Login';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  menu: {
    name: 'Test',
    routes: [],
  },
};

const routes = [
  {
    component: Home,
    routes: [
      {
        path: '/',
        exact: true,
        component: Home,
      },
      {
        path: '/login',
        component: Login,
      },
    ],
  },
];

const menuByHostResponse = {
  contains: [
    {
      route: '/login',
      roles: ['user'],
      _rev: '_Vzj_HTu---',
      entityId: 'e2',
      _id: 'menuItem/2',
      label: 'login',
      _key: '2',
    },
  ],
  _rev: '_Vzj_HRG---',
  host: 'example.org',
  description: 'example site111',
  entityId: 'e2',
  _id: 'menu/example.org',
  _key: 'example.org',
};

describe('Layout', () => {
  beforeEach(() => {
    window.fetch = jest
      .fn()
      .mockImplementation(() =>
        Promise.resolve(
          mockResponse(200, 'ok', JSON.stringify(menuByHostResponse)),
        ),
      );
  });

  test('renders children correctly', () => {
    const store = mockStore(initialState);
    const wrapper = renderer
      .create(<App context={{ store, routes }} />)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
