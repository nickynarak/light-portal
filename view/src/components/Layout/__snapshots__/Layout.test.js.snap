// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="ButtonAppBar-root-1"
    >
      <header
        className="MuiPaper-root-11 MuiPaper-shadow4-17 MuiAppBar-root-4 MuiAppBar-positionStatic-7 MuiAppBar-colorPrimary-9"
      >
        <div
          className="MuiToolbar-root-38 MuiToolbar-gutters-39"
        >
          <button
            aria-label="Menu"
            className="MuiButtonBase-root-49 MuiIconButton-root-40 MuiIconButton-colorContrast-42 ButtonAppBar-menuButton-2"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label-46"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-51 MuiIconButton-icon-47"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-52"
            />
          </button>
          <a
            className="ButtonAppBar-brand-3"
            href="/"
            onClick={[Function]}
          >
            <h2
              className="MuiTypography-root-59 MuiTypography-title-65 MuiTypography-colorInherit-78"
            >
              Test
            </h2>
          </a>
        </div>
      </header>
    </div>
  </div>
  <p>
    Welcome to light portal
  </p>
</div>
`;
